internal/productcatalog/feature_connector.go

```diff
@@ -118,13 +117,13 @@ func (c *featureConnector) ListFeatures(ctx context.Context, params ListFeatures
 	return c.featureRepo.ListFeatures(ctx, params)
 }
 
-func (c *featureConnector) GetFeature(ctx context.Context, featureID models.NamespacedID) (Feature, error) {
-	feature, err := c.featureRepo.GetByID(ctx, featureID)
+func (c *featureConnector) GetFeature(ctx context.Context, namespace string, idOrKey string) (*Feature, error) {
+	feature, err := c.featureRepo.GetByIdOrKey(ctx, namespace, idOrKey, true)
 	if err != nil {
-		return Feature{}, err
+		return nil, err
 	}
-	if feature.Namespace != featureID.Namespace {
-		return Feature{}, &FeatureNotFoundError{ID: featureID.ID}
+	if feature.Namespace != namespace {
```

Comment:
This is not needed as the GetByIdOrKey's idempotent property should be that the namespace is already filtered for. The previous implementation didn't have this check, so that's why it has been stuck here.